src/packages/core/file.cc

```diff
@@ -374,24 +374,52 @@ char *read_file(const char *file, int start, int lines) {
     return nullptr;
   }
   theBuff[total_bytes_read] = '\0';
-
-  // skip forward until the "start"-th line
-  const char *ptr_start = theBuff;
-  while (start > 1 && ptr_start < theBuff + total_bytes_read) {
-    if (*ptr_start == '\0') {
-      debug(file, "read_file: file contains '\\0': %s.\n", file);
-      return nullptr;
+  const char *ptr_start;
```

Comment:
I think you still want  const char *ptr_start  = theBuff; 

otherwise read_file(0, 1) will fail

Reply:
I have `    ptr_start = theBuff;` below that for if the line parameter is a positive number, and `ptr_start = theBuff + total_bytes_read;` if it is negative.